website/content/en/docs/advanced-topics/heterogeneous.md

```diff
@@ -0,0 +1,400 @@
+
+```yaml
+affinity:
+  nodeAffinity:
+    requiredDuringSchedulingIgnoredDuringExecution:
+      nodeSelectorTerms:
+      - matchExpressions:
+        - key: kubernetes.io/arch
+          operator: In
+          values:
+          - amd64
+          - arm64
+          - ppc64le
+          - s390x
+```
+
+The list of architecture values should only include the architectures the operator supports. The full list of possible `GOARCH` values is [available here](https://github.com/golang/go/blob/master/src/go/build/syslist.go). It should be noted that Kubernetes allows a user to specify both `requiredDuringSchedulingIgnoredDuringExecution` and `preferredDuringSchedulingIgnoredDuringExecution`. The syntax for each block is the same. At a minimum, an operator author should set the required terms since they protect against a pod being scheduled on an architecturally incompatible node. If the operator in question has much better performance on a subset of the supported architectures, it may also be prudent to set `preferredDuringSchedulingIgnoredDuringExecution` so that the optimal arches are selected first if they are available.
+
+#### Integrating with Operator Lifecycle Manager
+
+The Operator Lifecycle Manager (OLM) is often used to distribute operators via operator catalogs. The operator deployment object in an OLM integrated operator is defined in the `ClusterServiceVersion` yaml. It is important to remember to set the node affinity block in each of the `spec.template.spec` under `spec.install.deployments`.
+
+## Examples
+This section attempts to show how each of the sample operators could be updated to declare a set of supported architectures. Most of the time, these changes will be in `yaml` files, but different kinds of operators embed `PodSpec` and `PodTemplateSpec` objects in different places.
+
+The most common directory with objects that will need an architecture affinity block is the operator's `config` directory. Additionally, if the operator is configured for OLM, there will likely be a deployment object in the `bundle/manifests/*.clusterserviceversion.yaml`. These locations are common across all operator types.
+
+Each of these examples can be in the [OperatorSDK GitHub repository](https://github.com/operator-framework/operator-sdk).
+
+### Ansible Operators
+In the `testdata/ansible` samples app, you'll find several configuration objects that need a nodeAffinity specification in the `config` and `bundle/manifests` directories. The full list is as follows:
+- `testdata/ansible/memcached-operator/bundle/manifests/memcached-operator.clusterserviceversion.yaml`
+- `testdata/ansible/memcached-operator/config/default/manager_config_patch.yaml`
+- `testdata/ansible/memcached-operator/config/default/manager_auth_proxy_patch.yaml`
+- `testdata/ansible/memcached-operator/config/manager/manager.yaml`
```

Comment:
I do not think that we should use testdata/ansible. It does not exist when users create their projects with the tool.
The testdata is a dir that only exists in the SDK repo. Usually, we describe how things work in the project generated by the tool and we link the samples/examples or we say you can check it in the testdata/sample. 

However, provide the path `testdata/ansible/` is strange. 

Also, note that only the manager and manager_auth_proxy_patch.go (which will patch the manager to add the auth proxy container) are those that should be changed.

Reply:
I can certainly simplify the path.
One question I had when starting this is whether it would be more helpful to actually link the content. My worry is that if we link the content, it may change with updates over time that would break the links.

Can you explain to me what the `patch.yaml` file does so I can understand why it doesn't need the update?

Reply:
the patches do not define the container
Then, patch things such as add an envvar or an extra config.
See how to config and where the nodeAffinity by looking at the changes  proposed in: https://github.com/kubernetes-sigs/kubebuilder/pull/2906